### PR TITLE
x86_cpu_flags:add new case to test flag smap

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1855509

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test result:
 (1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.test_smap: PASS (42.84 s)

**Comment by nanliu-r** [comment]
@vivianQizhu @huangyum Please help to review this, thanks a lot.

File: qemu/tests/x86_cpu_flag_smap.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    2) For q35 machine type
+      2.1) Boot guest with supported cpu model
+      2.2) Check smap flag inside guest(only for linux guest)
+    3) For pc machine types
+      3.1) Check pc machine types that are newer than rhel7.0.0
+      3.2) Boot with new machine type
+      3.3) Check cpu flags inside guest(only for linux guest)
+      3.1) Check old pc machine types that are older than rhel7.0.0
+      3.2) Boot with old machine type
+      3.3) Check cpu flags inside guest(only for linux guest)
```
Comment (huangyum) [inline qemu/tests/x86_cpu_flag_smap.py:25]: Why are there two 3.1, 3.2, 3.3?

File: qemu/tests/x86_cpu_flag_smap.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        check_cpu_flags(params, flags, test)
+
+    boot_times = 1
+    machine_type = params['machine_type']
+    if machine_type == 'pc':
+        qemu_binary = utils_misc.get_qemu_binary(params)
+        query_cmd = "%s -M ?" % qemu_binary
+        qemu_binary_output = process.system_output(
+            query_cmd, shell=True).decode().splitlines()
+        pc_machine_types_new, pc_machine_types_old = get_machine_type(
```
Comment (huangyum) [inline qemu/tests/x86_cpu_flag_smap.py:59]: pc_machine_types_new  is useless?

**Review by huangyum — COMMENTED** [review]
Will you pls also provide test results on rhel7?

File: qemu/tests/x86_cpu_flag_smap.py (hunk 1 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_smap.py]: typo, thanks and update.

File: qemu/tests/x86_cpu_flag_smap.py (hunk 2 above)
Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_smap.py]: update and thanks.

File: qemu/tests/x86_cpu_flag_smap.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            query_cmd, shell=True).decode().splitlines()
+        pc_machine_types_old = get_machine_type(
+            qemu_binary_output)
+        if pc_machine_types_old:
+            boot_times = 2
+    for x in range(boot_times):
+        if x >= 1:
+            flags = ''
+            params['no_flags'] = 'smap'
+            params['machine_type'] = pc_machine_types_old[0]
```
Comment (vivianQizhu) [inline qemu/tests/x86_cpu_flag_smap.py:64]: You don't have the variant `pc_machine_types_old` when `params['machine_type']` is not `pc`, so this line you might hit error, right?

Comment (nanliu-r) [reply qemu/tests/x86_cpu_flag_smap.py:64]: > You don't have the variant `pc_machine_types_old` when `params['machine_type']` is not `pc`, so this line you might hit error, right?

If there is no `pc_machine_types_old` list the VM will only boot once. If there is `pc_machine_types_old` the VM will boot twice, the first time the machine type is pc and the second time the machine type is the latest in the `pc_machine_types_old` list.

Comment (vivianQizhu) [reply qemu/tests/x86_cpu_flag_smap.py:64]: @nanliu-r Okay, I see. Thanks for clarification.

**Comment by nanliu-r** [comment]
@huangyum @vivianQizhu Since we have no more test with RHEL7 and there is no machine type rhel6.x I drop some steps so there only need to test the flag `smap` on the RHEL8 product on which there is no rhel6.x machine type.
please review again, thanks.

 (1/2) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.9.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.test_smap: PASS (110.74 s)
 (2/2) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.x86_cpu_flags.others.test_smap: PASS (67.44 s)

**Comment by nanliu-r** [comment]
(1/1) Host_RHEL.m7.u9.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.x86_cpu_flags.others.test_smap: PASS (228.17 s)

**Review by huangyum — APPROVED** [review]
ACK

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**